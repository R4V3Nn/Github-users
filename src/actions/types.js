
export default {
	FETCH_USER: 'FETCH_USER',
	FETCH_USER_REPOS: 'FETCH_USER_REPOS',
	FETCH_NEXT_USER: 'FETCH_NEXT_USER',
	FETCH_PREV_USER: 'FETCH_PREV_USER',

	// LIST action types
	ITEM_TOGGLE: 'ITEM_TOGGLE',
};
