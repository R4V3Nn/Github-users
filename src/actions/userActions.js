import types from './types';

export const fetchUser = id => dispach => {
	fetch(`https://api.github.com/user/${id}`)
		.then(res => res.json())
		.then(data => dispach({
			type: types.FETCH_USER,
			payload: data,
		}));
};

export const fetchUserRepos = url => dispatch => {
	fetch(url)
		.then(res => res.json())
		.then(resp => dispatch({
			type: types.FETCH_USER_REPOS,
			payload: resp,
		}));
};

export const fetchRepos = (reposUrl, id) => dispatch => {
	if (!reposUrl) {
		fetch(`https://api.github.com/user/${id}`)
			.then(res => res.json())
			.then(data => dispatch(fetchUserRepos(data.repos_url)));
	} else {
		dispatch(fetchUserRepos(reposUrl));
	}
};

export const onToggle = id => ({
	type: types.ITEM_TOGGLE,
	payload: id,
});

export const formSubmit = id => ({
	types: types.FORM_SUBMIT,
	payload: id,
});
