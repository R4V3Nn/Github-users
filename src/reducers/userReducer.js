import types from '../actions/types';

const initialState = {
	user: {},
};

export default function (state = initialState, action) {
	switch (action.type) {
		case types.FETCH_USER:
			return {
				...state,
				user: action.payload,
			};
		case types.FETCH_USER_REPOS:
			return {
				...state,
				userRepos: action.payload,
			};
		case types.ITEM_TOGGLE:
			return {
				...state,
				selectedId: action.payload,
			};
		case types.FORM_SUBMIT:
			return {
			
			}
		default:
			return state;
	}
}

