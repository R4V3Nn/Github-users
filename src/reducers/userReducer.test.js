import userReducer from './userReducer';
import { FETCH_USER } from '../actions/types';

describe('userReducer', () => {
	test('is correct', () => {
		const action = { type: 'default_action' };


		expect(userReducer(undefined, action)).toMatchSnapshot();
	});
});

describe('FETCH_USER', () => {
	test('is correct', () => {
		const action = { type: FETCH_USER, payload: [] };


		expect(userReducer(undefined, action)).toMatchSnapshot();
	});
});
