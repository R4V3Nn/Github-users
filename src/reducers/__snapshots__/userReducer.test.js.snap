// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FETCH_USER is correct 1`] = `
Object {
  "filteredRepos": null,
  "selectedIds": Array [],
  "user": Object {},
}
`;

exports[`userReducer is correct 1`] = `
Object {
  "filteredRepos": null,
  "selectedIds": Array [],
  "user": Object {},
}
`;
